cdci_data_analysis/flask_app/app.py

```diff
@@ -392,6 +393,42 @@ def get(self, path):
                            e, status_code=410)
 
 
+@app.route('/post_product_to_gallery', methods=['POST'])
+def post_product_to_gallery():
+    logger.info("request.args: %s ", request.args)
+    logger.info("request.files: %s ", request.files)
+
+    token = request.args.get('token', None)
+    if token is None:
+        return make_response('A token must be provided.'), 403
+    try:
+        app_config = app.config.get('conf')
+        secret_key = app_config.secret_key
+        decoded_token = tokenHelper.get_decoded_token(token, secret_key)
+        logger.info("==> token %s", decoded_token)
+    except jwt.exceptions.ExpiredSignatureError:
+        # raise RequestNotAuthorized("The token provided is expired.")
+        return make_response('The token provided is expired.'), 403
+    except jwt.exceptions.InvalidTokenError:
+        # raise RequestNotAuthorized("The token provided is not valid.")
+        return make_response('The token provided is not valid.'), 403
+
+    jwt_pg_token = drupal_helper.get_mmoda_pg_token(app_config.product_gallery_jwt_token_location)
```

Comment:
so this token is limited lifetime, right? How do we update it in production? Did you manage to make it live longer? 
If not, we could also decode it with pyjwt, change, and encode back. If we can get the secret key from drupal (I assume it's possible).

In any case, would be nice to have a very simple endpoint to write this token. Since we will have to update it sooner or later.

Reply:
Investigation with @motame will perhaps help to better understand how to tackle this issue.

As also agreed, an  endpoint will also help.

Reply:
> Investigation with @motame will perhaps help to better understand how to tackle this issue.

Do try to decode the token, it's base64 json (part of it).

If it changes with new VPN, it might have host name or ip in it. Then it's no problem for production dispatcher, since it never changes of course.

> As also agreed, an endpoint will also help.

It will likely expire anyway. So an endpoint is critical. 
It will also help in short term no matter what are the other issues.

Reply:
Just verified that the validity of the token issued is 1 hour

Reply:
> Just verified: the validity of the token issued is 1 hour

if you have the secret key and the protocol, you might be able to produce (maybe easiest modify starting from existing) the token locally with `pyjwt`. And set whatever duration.

Reply:
> 
> if you have the secret key and the protocol, you might be able to produce (maybe easiest modify starting from existing) the token locally with `pyjwt`. And set whatever duration.

indeed what I am doing

Reply:
hm, it's really what I meant. I thought the token would be updated outside of dispatcher (with drupal or otherwise) and uploaded to it?

If we trust the dispatcher with the secret key, why not generate pg-token with every request? No need to even store it in a file?

Reply:
ok then in that case, there's no  need to take care of extending the validity of the token, am I right?

At every request a fresh new token will be generated, and the only secret information stored will be the key.

Reply:
yeah, if secret key is available, and tokens generated like make sense to drupal, this is sufficient. No need for an additional endpoint. 
We'll update the secret infrequently, and hence possible to do it on deployment.

Reply:
a new token is now generated at each request, and the secret_key is stored in the configuration